pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -289,6 +293,8 @@ func (p *DNSProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 	}
 	scopedLog.WithField("server", targetServerAddr).Debug("Found target server to of DNS request")
 
+	stat := spanstat.SpanStat{}
+	stat.Start()
```

Comment:
This might be slightly too soon? the CheckAllowed isn't part of the upstreaming, exactly, and it might be a non-trivial amount of time compared to talking to the upstream server. I'm also more worried about the `p.NotifyOnDNSMsg` when CheckAllowed succeeds. In that case we would include any time spent there, and that may block on a lock.
I'm not sure if we should add a different metric for the whole processing time, or change this one to that. They are two different concepts, though, and they measure different things (so, if you really want, let's have two).

Reply:
I think we want multiple spans so we can stack it to get the total time from the request received to the response sent